tests/softwareupdate/test_cloud_config.py

```diff
@@ -0,0 +1,668 @@
+                            },
+                            "develop": {
+                                "branch": "mrbeam2-stable",
+                                "branch_default": "mrbeam2-stable",
+                            },
+                            "alpha": {
+                                "branch": "mrbeam2-stable",
+                                "branch_default": "mrbeam2-stable",
+                            },
+                        },
+                        "2021-06-11": {
+                            "pip_command": "sudo /usr/local/netconnectd/venv/bin/pip",
+                            "stable": {"branch": "master", "branch_default": "master"},
+                        },
+                    },
+                }
+            },
+        },
+        "findmymrbeam": {
+            "name": "OctoPrint-FindMyMrBeam",
+            "repo": "OctoPrint-FindMyMrBeam",
+            "pip": "https://github.com/mrbeam/OctoPrint-FindMyMrBeam/archive/{target_version}.zip",
+            "restart": "octoprint",
+        },
+    },
+}
+
+
+@ddt
+class SettingsTestCase(unittest.TestCase):
+    _softwareupdate_handler = None
+    plugin = None
+
+    def setUp(self):
+        self.plugin = MrBeamPluginDummy()
+
+    def test_server_not_reachable(self):
+        plugin = self.plugin
+        tier = "DEV"
+        beamos_date = date(2018, 1, 12)
+        with requests_mock.Mocker() as rm:
+            rm.get(
+                "https://api.github.com/repos/mrbeam/beamos_config/tags",
+                json={"test": "test"},
+                status_code=404,
+            )
+            rm.get(
+                "https://api.github.com/repos/mrbeam/beamos_config/contents/docs/sw-update-conf.json?ref=vNone",
+                status_code=404,
+            )
+            tag_of_github_repo = get_tag_of_github_repo("beamos_config")
+            assert tag_of_github_repo == None
+            cloud_config = get_config_of_tag(tag_of_github_repo)
+            assert cloud_config == None
+            self.plugin._settings.set(tier)
+            update_config = _set_info_from_cloud_config(
+                plugin, tier, beamos_date, cloud_config
+            )
+            assert update_config == None
+
+    def test_cloud_confg_buster(self):
+        plugin = self.plugin
+
+        beamos_date = date(2021, 6, 11)
+        tiers = [
+            SW_UPDATE_TIER_DEV,
+            SW_UPDATE_TIER_ALPHA,
+            SW_UPDATE_TIER_BETA,
+            SW_UPDATE_TIER_PROD,
+        ]
+        # test for all tiers
+        for tier in tiers:
+            cloud_config = get_config_of_tag(get_tag_of_github_repo("beamos_config"))
+            update_config = _set_info_from_cloud_config(
+                plugin, tier, beamos_date, cloud_config
+            )
+            print("config {}".format(update_config))
+            assert (
+                update_config["octoprint"]
+                == target_octoprint_config[get_tier_by_id(tier)]
```

Comment:
seems like formating is not right or is it? you could use self.assertEquals https://docs.python.org/3/library/unittest.html#assert-methods